### ISSUE TITLE
Microphone: make echo test inherently optional and local (not E2E)

### ISSUE THREAD

**Issue by prlanzarin** [bigbluebutton/bigbluebutton#14716]
**Is your update request related to a problem? Please describe.**

There's not many good reasons to have an E2E echo test right now, IMO. Things I don't consider good arguments (and are frequently used as such):
 - _it serves as a "connectivity check"_: the actual audio connection establishment _is_ the connectivity test - that doesn't depend on E2E echo test;
 - _it would showcase E2E latency and quality_: the main thing about echo test is figuring out whether users' in/out devices are correctly setup, not gauging E2E latency/quality. That's not something they should worry about when opening a conference.

There's some other problems with it being E2E right now, mainly: it binds the audio negotiation flow with something very FreeSWITCH specific which is bridge transfers et al. A good chunk of technical debt and code could be cleared up (in multiple components) without E2E echo test.

Finally, it is currently a first class citizen in the join audio process - meaning it has a dedicated step in the audio join process (click join -> echo -> do you hear (y/n)? -> transfer). Due to that characteristic, the input device selection process is very _clunky_ (or impossible if echo is deactivated) if the user wants to get the devices right **before** joining audio. 
This process could be greatly simplified by, eg, unifying echo test with device selection.

**Describe the solution you'd like**

UX flow suggestion: open audio modal -> audio preview (input/output selectors, local echo/energy bar test) -> share microphone or cancel.

Tasks:
- [x] Bypass the E2E echo test process (dial to extension, validate, transfer) by making the feedback loop client-side only
- [x] Add an audio preview view to the audio join modal (re-use the current `AudioSettings` view - it's 2/3 there)
- [x] Add device selectors, any other needed options to that UI (re-use the current `AudioSettings` view - it's 2/3 there)
- [x] Add a local "echo test" of sorts to that preview UI (**which is NOT a first class citizen in the join audio process**)
   * Just picture the video-preview UI, but for audio, where input devices are mics and the preview is either an energy bar and/or an audible echo feedback, all client side.
- [x] Make the stuff introduced by the previous tasks **optional** and off by default (for starters)
  * Make sure everything we have today stays working in a first pass

**Describe alternatives you've considered**

- Keep echo E2E and just re-do the UI flow
- Remove echo test

**More**

This might be considered a duplicate of https://github.com/bigbluebutton/bigbluebutton/issues/10853.
I'm opening a new one so that the issue is specified with a task-oriented/technical perspective. Once a first prototype is made around _anything_ in this regard we can consider merging related issues.

Issues that this one may encompass:
  - https://github.com/bigbluebutton/bigbluebutton/issues/10853
  - https://github.com/bigbluebutton/bigbluebutton/issues/10757
